homeassistant/components/geocaching/sensor.py

```diff
@@ -1,74 +1,132 @@
 """Platform for sensor integration."""
 from __future__ import annotations
 
+from typing import Any
+
 from homeassistant.components.geocaching.coordinator import (
     GeocachingDataUpdateCoordinator,
 )
 from homeassistant.components.sensor import SensorEntity
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import ATTR_ICON
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.typing import HomeAssistantType
 
-from .const import (
-    ATTR_DEFAULT_ENABLED,
-    ATTR_DEVICE_CLASS,
-    ATTR_NAME,
-    ATTR_SECTION,
-    ATTR_STATE,
-    ATTR_UNIT_OF_MEASUREMENT,
-    DOMAIN,
-    SENSOR_ENTITIES,
-)
-from .models import GeocachingEntity
+from .const import DOMAIN
+from .models import GeocachingEntity, GeocachingSensorSettings
 
 
 async def async_setup_entry(
     hass: HomeAssistantType, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up a Geocaching sensor entry."""
 
+    sensor_data = {
```

Comment:
I think this actually should be in `const.py` as before.

Reply:
Circular dependency can be resolved using `typing.TYPE_CHECKING` constant.

This is clearly some hardcoded data which `const.py` is for.

Reply:
Can you elaborate on that, my python knowledge is not that advanced. if I wrap the .models import in the typing.TYPE_CHECKING  check it gives me an error about a missing import. I can't find out where I have to do that check.